src/client/datascience/notebook/helpers/helpers.ts

```diff
@@ -109,6 +109,11 @@ export function isPythonNotebook(metadata?: nbformat.INotebookMetadata) {
     if (metadata?.language_info?.name && metadata.language_info.name !== PYTHON_LANGUAGE) {
         return false;
     }
+
+    if (kernelSpec?.name.includes(PYTHON_LANGUAGE)) {
```

Comment:
This seems like it won't work consistently. What line here is causing the notebook to not be python otherwise?